numba/core/boxing.py

```diff
@@ -73,16 +73,31 @@ def box_float(typ, val, c):
 
 @unbox(types.Float)
 def unbox_float(typ, obj, c):
-    fobj = c.pyapi.number_float(obj)
-    dbval = c.pyapi.float_as_double(fobj)
-    c.pyapi.decref(fobj)
-    if typ == types.float32:
-        val = c.builder.fptrunc(dbval,
-                                c.context.get_argument_type(typ))
-    else:
-        assert typ == types.float64
-        val = dbval
-    return NativeValue(val, is_error=c.pyapi.c_api_error())
+    is_error_ptr = cgutils.alloca_once_value(c.builder, cgutils.false_bit)
+    float_ty = c.context.get_argument_type(typ)
+    float_val = cgutils.alloca_once(c.builder, float_ty)
+    with ExitStack() as stack:
+        fobj = c.pyapi.number_float(obj)
+        with early_exit_if_null(c.builder, stack, fobj):
```

Comment:
This would be slightly faster as the C code `if (dbval == -1 && PyErr_Occurred())`, but I have no idea how to write a short-circuiting `&&` in LLVM.

(from https://github.com/numba/numba/pull/5113#discussion_r935601482)